setup.py

```diff
@@ -23,62 +23,101 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import os
-from setuptools import setup
+import sys
+import logging
+
+from setuptools import setup, find_packages
 from setuptools.extension import Extension
-from Cython.Build import cythonize
-import Cython.Compiler.Options as CCO
 from setuptools.command.install import install
-import numpy as np          
 
-CCO.buffer_max_dims = 9
+logging.basicConfig(stream=sys.stderr, level=logging.DEBUG)
+log = logging.getLogger()
 
-class custom_install(install):
-    def run(self):
-        os.system("python setup.py build_ext --inplace -f")
-        install.run(self)
+# Courtesy of rasterio's setup.py.
+
+try:
+    import numpy as np
+except ImportError:
+    log.critical("Numpy and its headers are required to run setup(). Exiting.")
+    sys.exit(1)
+
+# Use Cython if available.
+
+try:
+    from Cython.Build import cythonize
+    import Cython.Compiler.Options as CCO
+except ImportError:
+    cythonize = None
+
+include_path = np.get_include()
```

Comment:
probably best to just make this an empty array if numpu is not installed

Reply:
The whole installation will bomb without numpy - I don't think the .c and .cpp files can compile without those numpy headers.

Reply:
opening, reading, parsing the setup.py file doesn't mean you are also actually installing it.